### PR TITLE
Wrap all tests in strictmode

### PR DESCRIPTION
**Pull request by tstirrat15**
## Motivation
This should help with development against future versions of React and enforces best practices generally.

## Changes
* Add test utils file with `renderStrict` and `serverRenderStrict` functions
* Implement those functions across all tests

## Testing
Code review. See that tests pass.

### CODE REVIEW

**Comment by tstirrat15** [comment]
Though one thing I'm curious about is that it seems that the arity of `renderStrict` is wrong, and yet tests still pass. Any ideas there?

I'm also not seeing a single warning on output, which seems strange.

**Comment by tstirrat15** [comment]
Mmmk. I wasn't calling the callback, which was one problem, and another problem is that the peer dependency specified in package.json is an old version of react which doesn't have strictmode, so the utils file is doing exactly as expected and making strictmode a passthrough component. I'll verify that the behavior is correct with a recent version of react and then we can figure out how we want to run tests against those versions of react.

**Comment by tstirrat15** [comment]
It seems I'll have to rewrite some of the tests as well. They aren't playing nicely with React 16. I'll try and get around to this on Monday.

**Comment by edorivai** [comment]
Thanks a lot! Looking forward to it.

File: modules/__tests__/Media-test.js [hunk 1]
```diff
@@ -77,34 +76,34 @@ describe('A <Media>', () => {
...
         const element = (
           <Media query="">
             <div>hello</div>
           </Media>
         );
 
-        ReactDOM.render(element, node, () => {
-          expect(node.firstChild.innerHTML || '').not.toMatch(/hello/);
+        renderStrict(element, node, () => {
+          expect(node.innerHTML || '').not.toMatch(/hello/);
```
Comment (tstirrat15) [inline modules/__tests__/Media-test.js]: `firstChild` was null, which meant that the statement was erroring out. Is this a good enough equivalent, or should it be `node.firstChild && node.firstChild.innerHtml || ''`?

File: modules/__tests__/Media-test.js [hunk 2]
```diff
@@ -117,8 +116,8 @@ describe('A <Media>', () => {
           />
         );
 
-        ReactDOM.render(element, node, () => {
-          expect(node.firstChild.innerHTML || '').not.toMatch(/hello/);
+        renderStrict(element, node, () => {
+          expect(node.innerHTML || '').not.toMatch(/hello/);
```
Comment (tstirrat15) [inline modules/__tests__/Media-test.js]: Same thing here.

**Comment by tstirrat15** [comment]
@edorivai this should be ready for review now. I left a couple of comments on the substantive changes I made to the tests.

It seems like the tests should ideally be run against multiple versions of React, unless the plan is to move to React 16 as a part of v2.

**Comment by edorivai** [comment]
> It seems like the tests should ideally be run against multiple versions of React, unless the plan is to move to React 16 as a part of v2.

That is actually a good point. Thing is, we plan to add a hooks API in 2.1, but this would be a separate bundle (like `react-media/hooks`). That hooks bundle would support `>=16.7`, but the main bundle could still support `>=15 || ^0.14.7`. Not sure what would be the correct version to require in the peerDependency in that case... :thinking:

File: modules/__tests__/Media-test.js [hunk 3]
```diff
@@ -77,34 +76,34 @@ describe('A <Media>', () => {
...
 
-        ReactDOM.render(element, node, () => {
+        renderStrict(element, node, () => {
           expect(node.firstChild.innerHTML).toMatch(/goodbye/);
         });
       });
     });
 
-    describe('and a render function', () => {
+    describe.only('and a render function', () => {
```
Comment (edorivai) [inline modules/__tests__/Media-test.js]:
```suggestion
    describe('and a render function', () => {
```
I suspect this `.only` is not intended to stay here.

File: modules/__tests__/Media-test.js (hunk 1 above)
Comment (edorivai) [reply modules/__tests__/Media-test.js]: I'm a bit confused. Do you mean this test was broken before your changes? Or did this break by adding `renderStrict`?
In any case, I think `node.innerHTML` should be fine.

**Comment by tstirrat15** [comment]
It wasn't broken before my changes. I think `renderStrict` did something, though it could also be the react 15->16 update if there were changes in how the react tree is committed to the DOM.

I'm more familiar with the [enzyme](https://airbnb.io/enzyme/docs/api/) method of writing tests, where it'd be pretty easy to say `.find('div').text().to.equal('hello')`, and the find statement will traverse and search the tree for you. Is there a way to do something similar in JSDOM? My concern is that this test wouldn't actually fail...

**Comment by edorivai** [comment]
I'm not that familiar with JSDOM myself, but it should be relatively straightforward to verify whether the test would fail, right? Let me know if you need help with verifying that!

**Comment by edorivai** [comment]
Yep, looks good to me. Thanks a lot!

**Pull request merged**